### PR TITLE
Change data provider

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

**Comment by amitaibu** [comment]
```php
  public function getQueryForList() {
    $query = parent::getQueryForList();

    $query->leftJoin();

    return $query;
  }
```

**Comment by amitaibu** [comment]
https://github.com/RESTful-Drupal/restful/pull/397/files

**Comment by shahardo** [comment]
Related #300

File: negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php [hunk 1]
```diff
@@ -50,10 +49,67 @@ public function publicFieldsInfo() {
...
+    // Add a query for meter_categories.
+    $field = field_info_field('field_meter_category');
+    $table_name = _field_sql_storage_tablename($field);
+    $query->leftJoin($table_name, 'cat', 'cat.entity_id=meter_nid');
+    $query->addField('cat', 'field_meter_category_target_id', 'meter_category');
+
+    return $query;
+  }
+
+  protected function parseRequestForListFilter() {
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php]: PHPdocs

File: negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php [hunk 2]
```diff
@@ -50,10 +49,67 @@ public function publicFieldsInfo() {
...
+
+    // Modify meter category filter
+    foreach ($filters as &$filter) {
+      // Not meter-category, skip.
+      if ($filter['public_field'] != 'meter_category') {
+        continue;
+      }
+
+      // Find meter-category taxonomy vid
+      $tax = taxonomy_vocabulary_get_names();
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php]: ``taxonomy_vocabulary_machine_name_load()``

File: negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php [hunk 3]
```diff
@@ -50,10 +49,67 @@ public function publicFieldsInfo() {
...
+      $tax_vid = $tax['meter_category']->vid;
+
+      // Modify the filter to find entities with meter-category as given
+      // in the filter, or any of its children
+      $term_ids = array();
+      foreach ($filter['value'] as $term_id) {
+        // Get meter category list of children.
+        $tree = taxonomy_get_tree($tax_vid, $term_id);
+        // The category itself is not in the tree, add it manually.
+        $term_ids[] = $term_id;
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/electricity/1.0/NegawattElectricityResource.class.php]: add the term ID as the key. or use array_unique()

**Pull request closed without merging**